### PR TITLE
parser,plan,ast: fix #4239, concatenates string literals which placed each other, and use first string as projection name

### LINKED ISSUE #4239: string literal compability
Please answer these questions before submitting your issue. Thanks!

1. What did you do?

`select "ss" "a";`

both TiDB and MySQL not set sql mode "ANSI_QUOTES"

2. What did you expect to see?

```
mysql> select "ss" "a";
+-----+
| ss  |
+-----+
| ssa |
+-----+
1 row in set (0.00 sec)
```

3. What did you see instead?

```
mysql> select "ss" "a";
+------+
| a    |
+------+
| ss   |
+------+
1 row in set (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V`)?

```
Release Version: 0.8.0
Git Commit Hash: 0d27b1f49e596a071b24e2a88ff9a3994910e5d0
Git Branch: master
UTC Build Time: 2017-08-18 11:00:58
```

### PR DESCRIPTION
**Pull request by winkyao**
#4239 

@tiancaiamao @hanfei1991 @zimulala PTAL

### CODE REVIEW

File: ast/expressions.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ var (
 // ValueExpr is the simple value expression.
 type ValueExpr struct {
 	exprNode
+	projectionName string
```
Comment (shenli) [inline ast/expressions.go]: Use an int offset instead of a string.

File: parser/lexer.go [hunk 2]
```diff
@@ -460,14 +460,9 @@ func startString(s *Scanner) (tok int, pos Pos, lit string) {
 	// See http://dev.mysql.com/doc/refman/5.7/en/string-literals.html
 	ch := s.skipWhitespace()
 	if s.sqlMode&mysql.ModeANSIQuotes > 0 &&
```
Comment (tiancaiamao) [inline parser/lexer.go]: return here.

File: ast/expressions.go (hunk 1 above)
Comment (winkyao) [inline ast/expressions.go]: Done! @shenli PTAL

**Comment by tiancaiamao** [comment]
LGTM

File: parser/lexer.go [hunk 3]
```diff
@@ -451,19 +451,7 @@ func scanQuotedIdent(s *Scanner) (tok int, pos Pos, lit string) {
...
-
-	// Quoted strings placed next to each other are concatenated to a single string.
-	// See http://dev.mysql.com/doc/refman/5.7/en/string-literals.html
-	ch := s.skipWhitespace()
-	if s.sqlMode&mysql.ModeANSIQuotes > 0 &&
-		(ch == '"' || s.r.s[pos.Offset] == '"') {
-		return
-	}
-	return
+	return s.scanString()
```
Comment (shenli) [inline parser/lexer.go]: Seems that the startString() is useless now.

**Pull request merged**